integration-tests/connect_test.go

```diff
@@ -45,6 +46,40 @@ func TestGoCqlConnect(t *testing.T) {
 	require.Equal(t, "fake", iter.Columns()[0].Name)
 }
 
+func TestProtocolVersionNegotiation(t *testing.T) {
+	testCassandraVersion := env.CassandraVersion
+	env.CassandraVersion = "2.1" // downgrade C* version for protocol negotiation test
+	defer func() {
+		env.CassandraVersion = testCassandraVersion
+	}()
+	c := setup.NewTestConfig("", "")
+	c.ProtocolVersion = 4 // configure unsupported protocol version
+	testSetup, err := setup.NewSimulacronTestSetupWithConfig(t, c)
+	require.Nil(t, err)
+	defer testSetup.Cleanup()
+
+	// Connect to proxy as a "client"
+	proxy, err := utils.ConnectToClusterUsingVersion("127.0.0.1", "", "", 14002, 3)
+
+	if err != nil {
+		t.Log("Unable to connect to proxy session.")
+		t.Fatal(err)
```

Comment:
why not `t.Fatal("Unable to connect to proxy session: ", err)` just like you did a few lines below